crates/astria-composer/src/executor/mod.rs

```diff
@@ -553,6 +590,22 @@ impl Future for SubmitFut {
                     Ok(rsp) => {
                         let tendermint::abci::Code::Err(code) = rsp.code else {
                             info!("sequencer responded with ok; submission successful");
+
+                            metrics::gauge!(crate::metrics_init::CURRENT_NONCE)
```

Comment:
If I read this right `CURRENT_NONCE` is only set after the `WaitingForSend` fut completes (or fails). IMO it should be set at the beginning of each attempt to submit a future.

Please move the emission of this metric into `submit_tx` - that way you will get the gauge also after `WaitingForNonce` completes and a new submission is attempted.